dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java

```diff
@@ -795,6 +797,10 @@ public void setExporter(Exporter<T> exporter) {
 
         @Override
         public void unexport() {
+            if (unexported) {
```

Comment:
It would be better to use `AtomicBoolean`, and use cas to set ture here to prevent concurrency problem

Reply:
Thanks, according to your suggestion, it has been adjusted to AtomicBoolean type